packages/core/src/tools/timeUtils.ts

```diff
@@ -0,0 +1,44 @@
+import { isNumber, round } from './utils'
+
+export type Duration = number & { d: 'Duration in ms' }
+export type ServerDuration = number & { s: 'Duration in ns' }
+export type TimeStamp = number & { t: 'Epoch time' }
+export type RelativeTime = number & { r: 'Time relative to navigation start' } & { d: 'Duration in ms' }
+
+export function toOptionalServerDuration(duration: Duration | undefined): ServerDuration | undefined {
+  if (!isNumber<Duration>(duration)) {
+    return duration
+  }
+  return toServerDuration(duration)
+}
+
+export function toServerDuration(duration: Duration) {
+  return round(duration * 1e6, 0) as ServerDuration
+}
```

Comment:
Previous type strategy was not working since "normal" input and output types are now different
```ts
export function msToNs<T>(duration: number | T): number | T 
```
Do you see any trick to have a single function?

Reply:
Maybe something like this?
```ts
export function toOptionalServerDuration(duration: Duration | undefined): ServerDuration | undefined {
  if (!isNumber<Duration>(duration)) {
    return duration
  }
  return toServerDuration(duration)
}
export function toServerDuration(duration: Duration): ServerDuration
export function toServerDuration(duration: undefined): undefined
export function toServerDuration(duration: Duration | undefined) {
  if (!isNumber(duration)) {
    return undefined
  }
  return round(duration * 1e6, 0) as ServerDuration
}
```